Always use the demo config when outputting a new config

jsternberg opened the pull request:
Fixes #6968.

nathanielc commented:
Can we log here https://github.com/influxdata/influxdb/blob/e37bf55e32ab600bb925adb85ea417873fcd0265/cmd/influxd/run/config_command.go#L40 which config file was loaded?

jsternberg commented:
I've updated it to print to standard error when it merges the configuration files and I've also updated it so specifying `/dev/null` will skip the standard loading locations.

nathanielc commented:
LGTM :+1:
